pkg/backup/backupinfo/manifest_handling.go

```diff
@@ -1671,6 +1662,57 @@ func (f *IterFactory) NewDescIter(ctx context.Context) bulk.Iterator[*descpb.Des
 	return newSlicePointerIterator(f.m.Descriptors)
 }
 
+// Close closes the iterator.
```

Comment:
thoughts on moving the `DescIterator` and its initializer over to `desc_sst.go`? This would align with the file iterator living in `file_sst.go` and the descriptor revision iterator in `desc_sst.go`

Reply:
There are iterator tests in TestManifestHandlingIteratorOperations. They look good enough that I don't think we need a new set of tests.